test/service_objects/game_piece_action_resolvers/move_resolver.test.ts

```diff
@@ -124,28 +156,54 @@ describe('resolveMoveAction', () => {
       rangedWoundRoll: 4,
       rangedArmorPiercing: 1,
       rangedDamage: 1,
-      rangedAmmo: 5
+      rangedAmmo: 5,
     });
 
     let enemyUnit = await Factories.createUnit();
-    let movingPlayer = await Factories.createPlayer();
-    let enemyPlayer = await Factories.createPlayer();
-    let movingPlayerUnit = await Factories.createPlayerUnit(movingPlayer, movingUnit);
-    let enemyPlayerUnit = await Factories.createPlayerUnit(enemyPlayer, enemyUnit);
+    let movingPlayer = await Factories.createPlayer(p1PrivateKey.toPublicKey());
+    let enemyPlayer = await Factories.createPlayer(p2PrivateKey.toPublicKey());
+    let movingPlayerUnit = await Factories.createPlayerUnit(
+      movingPlayer,
+      movingUnit
+    );
+    let enemyPlayerUnit = await Factories.createPlayerUnit(
+      enemyPlayer,
+      enemyUnit
+    );
     let movingGamePlayer = await Factories.createGamePlayer(game, movingPlayer);
     let enemyGamePlayer = await Factories.createGamePlayer(game, enemyPlayer);
 
-    movingGamePiece = await Factories.createGamePiece(movingGamePlayer, movingPlayerUnit, 10, 10);
-    enemyGamePiece = await Factories.createGamePiece(enemyGamePlayer, enemyPlayerUnit, 10, 15);
+    movingGamePiece = await Factories.createGamePiece(
+      movingGamePlayer,
+      movingPlayerUnit,
+      10,
+      10
+    );
+    enemyGamePiece = await Factories.createGamePiece(
+      enemyGamePlayer,
+      enemyPlayerUnit,
+      10,
+      15
+    );
 
     let gamePhase = await Models.GamePhase.create({
       gameId: game.id,
       gamePlayerId: movingGamePlayer.id,
       turnNumber: 1,
-      phase: 'movement'
+      phase: 'movement',
     });
 
     let currentPos = movingGamePiece.coordinates();
+
+    const snarkyAction = new Action(
+      Field(1),
+      Field(0),
+      Position.fromXY(currentPos.x + 5, currentPos.y).hash(),
+      Field(await movingGamePiece.gamePieceNumber())
+    );
+    const signature = snarkyAction.sign(p1PrivateKey);
```

Comment:
Simulates the frontend - a user signs their move with their private key, and it must agree with the data that the server comes up with as proof input.

The move resolver does not need access to the user's public key - we will eventually get this signature directly in the graphql mutation.